### PR TITLE
feat: publish pubsub events

### PR DESCRIPTION
**Pull request by matt2e**
- pipes module calls for publishing events through to the controller to the db
- fake implementation for tests not included
- added TopicEventKey. It will be generally useful I think, but also required for how db events are set up needing a "key" column

### CODE REVIEW

File: go-runtime/internal/impl.go [hunk 1]
```diff
@@ -41,3 +41,20 @@ func (r *RealFTL) FSMSend(ctx context.Context, fsm, instance string, event any)
 	}
 	return nil
 }
+
+func (r *RealFTL) PublishEvent(ctx context.Context, topic string, event any) error {
+	fmt.Printf("impl: publish event")
```
Comment (alecthomas) [inline go-runtime/internal/impl.go:46]: Printf

File: go-runtime/internal/impl.go [hunk 2]
```diff
@@ -41,3 +41,20 @@ func (r *RealFTL) FSMSend(ctx context.Context, fsm, instance string, event any)
...
+	client := rpc.ClientFromContext[ftlv1connect.VerbServiceClient](ctx)
+	body, err := encoding.Marshal(event)
+	if err != nil {
+		return fmt.Errorf("failed to marshal event: %w", err)
+	}
+	_, err = client.PublishEvent(ctx, connect.NewRequest(&ftlv1.PublishEventRequest{
+		Topic: &schemapb.Ref{Module: reflection.Module(), Name: topic},
```
Comment (alecthomas) [inline go-runtime/internal/impl.go:53]: What if you're publishing to a module that's from another module?

**Review by alecthomas — APPROVED** [review]
Nice

File: go-runtime/internal/impl.go (hunk 2 above)
Comment (matt2e) [reply go-runtime/internal/impl.go:53]: Good point, I've made it an issue and part of the pubsub epic
https://github.com/TBD54566975/ftl/issues/1640

**Pull request merged**